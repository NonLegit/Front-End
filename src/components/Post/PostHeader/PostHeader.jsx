// mui components
import {
  Avatar, Box, Typography,
} from '@mui/material';
import CallSplitIcon from '@mui/icons-material/CallSplit';
import calculateTime from '../../../utils/calculateTime';

// styles
import {
  CreatedAt,
  Flair,
  PostInfo, PostInfoLink, PostTitle, Tag,
} from './styles';
/**
 * This component is the upper section of post
 *
 * @component PostHeader
 * @property {string} title -Post title.
 * @property {string} ownerType -Post owner type user or subreddit.
 * @property {string} ownerIcon -Post ownerName icon.
 * @property {string} ownerName -Post subreddit(post ownerName).
 * @property {string} authorName -Post authorName.
 * @property {string} flairText -Post flair text.
 * @property {string} flairBackgroundColor -Post flair background color.
 * @property {string} flairColor -Post flair color.
 * @property {boolean} subredit -to identify if post in home page or subreddit.
 * @property {boolean} nsfw -Whether the post is not safe for work or not.
 * @property {boolean} spoiler -Whether the post is spoiler or not.
 * @returns {React.Component} Post header
 */

function PostHeader(props) {
  const {
    title, ownerIcon, ownerType, ownerName, authorName, flairText, flairBackgroundColor, flairColor, createdAt,
    subredit, nsfw, spoiler, redirectToPost, sharedFrom,
  } = props;

  const handleClickOnTitle = (e) => {
    e.preventDefault();
    redirectToPost(true, false);
  };

  return (
    <>
      <PostInfo pb={1}>
        {!subredit
        && (
          <>
            <Avatar
              src={ownerIcon}
              sx={{
                width: 20,
                height: 20,
              }}
              alt="Profile ownerIcon"
            />
            <PostInfoLink to={ownerType === 'Subreddit' ? `/r/${ownerName}` : `/user/${ownerName}`} color="#000" fontWeight="bolder">
              {ownerType === 'Subreddit' ? 'r/' : 'u/'}
              {ownerName}
            </PostInfoLink>
          </>
        )}
        <Box color="#787C7E" fontWeight={300} display="flex" gap="4px" flexWrap="wrap">
          {ownerType === 'Subreddit' && (
          <>
            {!subredit && (
            <span>
              â€¢
            </span>
            )}
            {sharedFrom
              ? (
                <>
                  <CallSplitIcon
                    sx={{
                      color: '#0079D3',
                      transform: 'rotate(90deg)',
                      width: 15,
                      height: 15,
                    }}
                  />
                  <div>Crossed By</div>
                </>
              )
              : <div>Posted By</div>}
            <PostInfoLink to={`/user/${authorName}`} color="inherit" fontWeight="normal">
              u/
              {authorName}
            </PostInfoLink>
          </>
          )}

          <CreatedAt color="inherit" fontWeight="normal">
            {calculateTime(createdAt)}
          </CreatedAt>
        </Box>
      </PostInfo>
      <PostTitle onClick={handleClickOnTitle}>
        <Typography
          variant="h6"
          component="h3"
          sx={{ fontSize: 18 }}
        >
          {title}
          {' '}
          {
            flairText
            && (
            <Flair
              disableRipple
              backgroundColor={flairBackgroundColor}
              flairColor={flairColor}
            >
              {flairText}
            </Flair>
            )
          }
          {' '}
          {
            spoiler && (
            <Tag
              color="third"
              variant="outlined"
            >
              spoiler
            </Tag>
            )
          }
          {
            nsfw && (
            <Tag
              color="nsfw"
              variant="outlined"
            >
              nsfw
            </Tag>
            )
          }
        </Typography>
      </PostTitle>
    </>
  );
}

export default PostHeader;
