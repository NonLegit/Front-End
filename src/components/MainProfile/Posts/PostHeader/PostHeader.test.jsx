import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
// import MainProfile from '../../MainProfile';
import PostHeader from './PostHeader';

test('test snapshot', async () => {
  const tree = renderer.create(
    <Router>
      {/* <MainProfile> */}
      <PostHeader />

      {/* </MainProfile> */}

    </Router>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
