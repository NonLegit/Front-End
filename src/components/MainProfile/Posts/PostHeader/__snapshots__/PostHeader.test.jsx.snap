// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  className="MuiBox-root css-1jbw2pj"
>
  <div
    className="MuiBox-root css-2imjyh"
  >
    <div
      className="MuiBox-root css-14fy6pd"
    >
      <img
        className="css-1t9pz9x"
      />
    </div>
    <a
      className="css-4gr578"
      href="/user/undefined"
      onClick={[Function]}
    >
      <span
        className="MuiTypography-root MuiTypography-caption css-2w43o6-MuiTypography-root"
      >
        u/
         
        .
         
      </span>
    </a>
    <span
      className="MuiTypography-root MuiTypography-caption css-o4wjv1-MuiTypography-root"
    >
       
      Posted by
    </span>
    <a
      className="css-4gr578"
      href="/user/undefined"
      onClick={[Function]}
    >
      <span
        className="MuiTypography-root MuiTypography-caption css-auilwn-MuiTypography-root"
      >
        u/
      </span>
    </a>
    <div
      className="MuiBox-root css-dtw3z"
    >
      <span
        className="MuiTypography-root MuiTypography-caption css-o4wjv1-MuiTypography-root"
      >
         
        a few seconds ago
      </span>
    </div>
  </div>
</div>
`;
